### PR TITLE
Add header and impl to modifiers enum

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/KModifier.kt [hunk 1]
```diff
@@ -61,7 +61,11 @@ enum class KModifier(
   // Type modifiers.
   IN("in", Target.VARIANCE_ANNOTATION),
   OUT("out", Target.VARIANCE_ANNOTATION),
-  VARARG("vararg", Target.PARAMETER);
+  VARARG("vararg", Target.PARAMETER),
+
+  // Multiplatform modules.
+  HEADER("header", Target.CLASS),
+  IMPL("impl", Target.CLASS);
```
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/KModifier.kt]: This is also valid in a function and probably a property

Comment (Egorand) [reply src/main/java/com/squareup/kotlinpoet/KModifier.kt]: Is there any documentation btw?

**Comment by Egorand** [comment]
These modifiers are not mentioned in https://github.com/yole/kotlin-style-guide/issues/3

File: src/main/java/com/squareup/kotlinpoet/KModifier.kt (hunk 1 above)
Comment (JakeWharton) [reply src/main/java/com/squareup/kotlinpoet/KModifier.kt]: Uhh. I don't think there's much if any. I can point you to a bunch of example usage though.

Comment (Egorand) [reply src/main/java/com/squareup/kotlinpoet/KModifier.kt]: Sure, that'd be great

Comment (JakeWharton) [reply src/main/java/com/squareup/kotlinpoet/KModifier.kt]: https://github.com/JakeWharton/Reagent has the referenced atomic thing from the issue.

The others are private. Can invite later when I'm not mobile. Beyond that you can search "Kotlin multiplatform Gradle" and find a few others' samples.

Comment (Egorand) [reply src/main/java/com/squareup/kotlinpoet/KModifier.kt]: Thanks!

**Comment by JakeWharton** [comment]
Damn I forgot to squash

**Pull request merged**